pkg/pillar/cmd/upgradeconverter/applydefaultconfigitem.go

```diff
@@ -39,6 +46,12 @@ func applyDefaultConfigItem(ctxPtr *ucContext) error {
 		log.Noticef("No existing ConfigItemValueMap; creating %s with defaults",
 			newConfigItemFile)
 	}
+	keyData := readAuthorizedKeys(baseAuthorizedKeysFile)
+	if len(keyData) != 0 {
+		log.Functionf("Found the key data in %s", baseAuthorizedKeysFile)
+		newConfigPtr.SetGlobalValueString(types.SSHAuthorizedKeys, keyData)
+		defer os.Remove(baseAuthorizedKeysFile)
```

Comment:
I wonder if this should be more  bullet proof by saving the file to delete, and only delete it once the newConfig has been published. (I also wonder whether we should add some sync() calls but maybe that belongs in the generic pubsub code for persistent publications.)

Reply:
The call to fileutils.WriteRename below (in the same function) does Sync call to make sure that the json file is written to disk. This file is only deleted after the auth key is written to /persist/status/zedagent/ConfigItemValueMap/global.json

I have added the sync call inside WriteRename below.

Reply:
OK, but still shouldn't do the os.Remove unless the code path has actually done the publish of the ConfigItemValueMap so that we don't risk loosing the authorized key in some error path.

Reply:
Ok. Will move the file remove call to the bottom of function. That way the file is only delete in the non-error case.